# Chapter 4 - Discounted Cash Flow Valuation

**Chapter's Examples**

Corporate Finance 9th International Edition

Ross, Westerfield, Jaffe

In [20]:
import pandas as pd
import numpy as np
import numpy_financial as npf
from IPython.display import display, HTML

## Example 4.1 - Present Value

Linda Jennings, a  financial analys at kaufman & Broad, a leading real estate firm, is thinking about recommending that Kaufman & Broad invest in a piece of land that costs 85,000.  She is certain that next year the land will be worth 91,000, a sure 6,000 gain.  Given that the guaranteed interest rate in the bank is 10 percent, should Kaufman & Broad undertake the investment in land?

A moment's though should be all it takes to convice her that this is not an attractive business deal.  By investing 85,000 in the land, she will have 91,000 available next year.  Suppose instead, that Kaufman & Broad puts the same 85,000 into the bank.  At the interest rate of 10 percent, this 85,000 would grow to:

In [16]:
cost = 85000
fv = (1 + 10) * cost 
print(f' ( 1 + .10 ) x ${cost:,.0f} = ${fv:,.0f} next year.')

 ( 1 + .10 ) x $85,000 = $935,000 next year.


It would be foolish to buy the land when investing the same 85,000 in the financial market would produce an extra 2,500 (that is, 93,500 from the bank minus 91,000 from the land investment).  This is a future value calculation.

Alternatively, she could calculate the present value of the sale price next year as:

In [17]:
rate = .10
land_fv = 91000
land_pv = land_fv / (1 + rate)
print(f'Present Value = {land_fv:,.0f} / ( 1 + rate ) = {pv:,.0f}')

Present Value = 91,000 / ( 1 + rate ) = 384,000


Because the present value of next year's sales price is less than this year's purchase price of 85,000, present value analysis also indicates that she should not recommend purchasing the property.

## Example 4.2 - Uncertainty and Valuation

Profesional Artworks, Inc., is a firm that speculates in modern paintings.  The manager is thinking of buying an original Picasso of 400,000 with the intention of selling it at the end of one year.  The manager expects that the painting will be worth 480,00 in one year. 

Of course, this is only an expectation, the painting could be worth more or less that 480,000.  Suppose the guaranteed interest rate granted by banks is 10 percent.  Should the firm purchase the piece of art?

Our first thought might be to discount at the interest rate, yielding:

In [19]:
rate = .10
art_fv = 480000
art_pv = art_fv / (1 + rate) 
print(f'${art_fv:,.0f} / 1.10 = ${art_pv:,.0f}')

$480,000 / 1.10 = $436,364


Because 436,364 is greater thatn 400,000, it looks at first glance as if the painting should be purchased.  However, 10 percent is the return one can earn on a riskless investment.  Because the painting is quite risky, a higher discount rate is called for.  The manager chooses a rate of 25 percent to reflect this risk.  In other workds, he argues that a 25 percent expected return is fair compensation for an investment as risky as this painting.

The present value of the painting becomes:



In [20]:
rate = .25
art_fv = 480000
art_pv = art_fv / (1 + rate) 
print(f'${art_fv:,.0f} / 1.10 = ${art_pv:,.0f}')

$480,000 / 1.10 = $384,000


Thus, the manager believes that the painting is currently overpriced at 400,000 and does not make the purchase.

## Example 4.3 - Interest on Interest

Suh-Pyng Ku has put 500 in a savings account at the First National Bank of Kent.  The account earns 7 percent, compounded annually.  How much will Ms. Ku have at the end of three years?  The answer is:

In [25]:
rate = .07
deposit = 500
fv = deposit * (1 + rate)**3
print(f'= ${deposit:,.0f} x 1.07 x 1.07 x 1.07 \n= ${deposit:,.0f} x (1.07)^3 = ${fv:,.2f}')

= $500 x 1.07 x 1.07 x 1.07 
= $500 x (1.07)^3 = $612.52


## Example 4.4 - Compound Growth

Jay Ritter invested 1,000 in the stock of the SDH Company.  The company pays a currend dividend of 2, which is expected to grow by 20 percent per year for the next two years.  What will the dividend of the SDH Company be after two years?  A simple calculation gives:

In [28]:
div = 2
years = 2
growth_r = .20

future_div = div * (1 + growth_r)**years 
print(f'{div:.0f} x (1.20)^2 = ${future_div:.2f}')

2 x (1.20)^2 = $2.88


## Example 4.5 - Finding the Rate

Carl Voigt, who recently won 10,000 in the lottery, wants to buay a car in five years.  Carl estimates that the car will cost 16,105 at that time.  What interest rate must he earn to be able to afford the car?

The ratio of purchase price to initial cash is:

In [30]:
fv = 16105
pv = 10000
ratio = fv / pv
print(f'${fv:,.0f} / ${pv:,.0f} = {ratio:.4f}')

$16,105 / $10,000 = 1.6105


Thus, he must earn an interest rate that allows 1 to become 1.6105 in five years.  We can express the problem algebraically as:

In [32]:
print(f'${pv:,.0f} x (1 + r)^5 = {fv:,.0f}')

$10,000 x (1 + r)^5 = 16,105


where r is the interest rate needed to purchase the car.  Because 16,105/10,000 = 1.6105, we have:

In [40]:
r = ratio**(1/5) - 1
print(f'(1 + r)^5 = {ratio:.4f}\nr = {r*100:.0f}%')

(1 + r)^5 = 1.6105
r = 10%


Either a table, a spreadsheet, or a hand calculator lets us solve for r. (or python)

In [35]:
rate = npf.rate(pv=-pv, fv=fv, nper=5, pmt=0)
print(f'r = {rate*100:.0f}%')

r = 10%


## Example 4.6 - How Much for That Island?

Some people have said that it was the best real state deal in history.  Peter Minuit, director general of New netherlands, the Dutch West India Company's colony in North America, in 1626, allegedly bought Manhattan Island for 60 guilders' worth of trinkets from native Americans.  By 1667, the Dutch were forced by the British to exchange it for Suriname (perhaps the worst real estate deal ever).  This sounds cheap; but did the Dutch really get the better end of the deal?  It is reported that 60 guilders was worth about 24 dollars at the prevailling exchange rate.  If the native Americans had sold the trinkets at a fair market value an invested the 24 dollars at 5 percent (tax free), it would now, abut 383 years later, be worth more that 3.1 billion.  Today, Manhattan is undoubtedly worth more than 3.1 billion, so at a 5 percent rate of return the native Americans got the worst of the deal.  However, if invested at 10 percent, the amount of money thy received would be worth about:

In [42]:
r = .10
n = 383
pv = 24
fv = pv * (1 + r)**n
print(f'=${pv:.0f}(1 + r)^T = {pv:.0f} x (1 + .{r*100:.0f})^{n} \n= ${fv:,.0f}')

=$24(1 + r)^T = 24 x (1 + .10)^383 
= $171,241,749,947,655,040


This is a lot of money.  In fact, $171 quadrillion is more that all the real estate in the world is worth today.  Note that no one in the history of the world has ever been able to find an investment yielding 10 percent every year for 383 years.

## Example 4.7 - Multiperiod Discounting

Bernard Durmas will receive 10,000 three years from now.  Bernard can earn 8 percent on his investments, so the appropriate discount rate is 8 percent.  What is the present value of his future cash flow?  The answer is:

In [46]:
r = .08
n = 3
fv = 10000
pv_fact = 1 / (1 + r)**n
pv = fv * pv_fact
print(f'PV = ${fv:,.0f} x (1 / (1 + {r})^3\nPV = {fv:,.0f} x {pv_fact:.4f}\nPV = {pv:,.0f}')

PV = $10,000 x (1 / (1 + 0.08)^3
PV = 10,000 x 0.7938
PV = 7,938


When his investments grow at an 8 percent rate of interest, Bernard Durmas is equally inclined toward receiving 7,938 now and receiving 10,000 in three years' time.  After all, he could convert the 7,938 he receives today into 10,000 in three years by lending it at an interest rate of 8 percent.

## Example 4.8 - Finding the Rate

A customer of the Chaffkin Corp. wans to buy a tugboat today.  Rather than paying inmediately, he will pay 50,000 in three years.  it will cost the Chaffkin Corp. 38,610 to build the tugboat immediately.  What interest rate would the Chaffkin Corp. charge to neither gain nor lose on the sale?

In [47]:
n = 3
pv = 38610
fv = 50000
ratio = pv / fv
print(f'The ratio of construction cost to sale price is: ${pv:,.0f} / ${fv:,.0f} = {ratio:.4f}')

The ratio of construction cost to sale price is: $38,610 / $50,000 = 0.7722


We must determine the interest rate that allows 1 dollar to be received in three years to have a present value of .7722 dollars.

## Example 4.9 - Waiting for Godot

You've been saving up to buy the Godot Company.  The total cost will be 10 million.  You currently have about 2.3 million.  If you can earn 5 percent on your mony, how log will you have to wait? At 16 percent, how long must you wait?

At 5 percent, you'll have to wait a long time.  For the basic present value equation:

2.3 million = 10 million / 1.05^t

1.05^t = 4.35

t = 30 years

At 16 percent, things are a little better.  Verify for yourself that it will take about 10 years.

In [54]:
r = .16
fv = 10000000 
pv = 2300000
n = npf.nper(pv=-pv, fv=fv, rate=r, pmt=0)
print(f'It will take approximately {n:.0f} years.')

It will take approximately 10 years.


## Example 4.10 - Cash Flow Valuation

Kyle Mayer has won the Kentucky State Lottery and will receive the following set of cash flows over the next two years:

1. 20,000
2. 50,000

Mr. Mayer can currently earn 6 percent in his money market account, so the appropriate discount rate is 6 percent.  The prsent value of the cash flows is:

In [1]:
r = .06
cf = [0, 20000, 50000]
pv_of_cash_flows = 0

print("Year", "Cash_Flow", "PV_Factor", "Present_Value", sep=" "*4)

for k, v in enumerate(cf[1:], start=1):
    pv_fact = (1 / (1 + r))**(k)
    pv = v * pv_fact
    pv_of_cash_flows += pv
    print(f"   {k}     {v:,.0f}       {pv_fact:.4f}        {pv:,.2f}")

print(f"Total                               {pv_of_cash_flows:,.2f}")


Year    Cash_Flow    PV_Factor    Present_Value
   1     20,000       0.9434        18,867.92
   2     50,000       0.8900        44,499.82
Total                               63,367.75


In other words, Mr. Mayer is equally inclined toward receiving 63,367.75 today and receiving 20,000 and 50,000 over the next two years.

In [4]:
# Manual PV calculation shown above for illustration purposes. 
# npf.npv could be used instead.

pv_of_cf = npf.npv(r, cf)
print(f"${pv_of_cf:,.2f}")

$63,367.75


## Example 4.11 - Net Present Value

Finance.com has an opportunity to invest in a new high-speed computer that costs 50,000.  The computer will generate cash flows (from cost savings) of 25,000 on year from now, 20,000 two years from now, and 15,000 three years from now.  The computer will be worthless after three years, and no additional cash flows will occur.  Finance.com has determined that the appropriate discount rate is 7 percent for this investment.  Should finance.com make this investment in a new high-speed computer?  What is the net present value of the investment?

The cash flows and present valu factors of the proposed computer are s follow:

In [ ]:
"""
             Cash_Flow             PV_Factor
Year 0        -50,000                1 = 1
Year 1         25,000           1/1.07 = .9346
Year 2         20,000       (1/1.07)^2 = .8734
Year 3         15,000       (1/1.07)^3 = .8163
"""

The present value of the cash flows is:

Cash flows x Present Value Factor = Present Value

In [38]:
r = .07

# Create a basic data structure
struct = {"Years": [0, 1, 2, 3],
          "Cash_Flows": [-50000, 25000, 20000, 15000],
          "PV_Factor": [],
          "Present_Value": []
         }

# Manually discount Cash Flows 
# to get Pv_factors and present values
for t in struct["Years"]:
    if t == 0:
        pv_fact = 1
        pv = struct["Cash_Flows"][t] * pv_fact
    else:
        pv_fact = 1 / (1 + r)**t
        pv = struct["Cash_Flows"][t] * pv_fact

    struct["PV_Factor"].append(pv_fact)
    struct["Present_Value"].append(pv)

# Create table (dataframe)
df = pd.DataFrame(struct)

# Add a total row
total_row = pd.DataFrame({
    "Years": ["Totals"],
    "Cash_Flows": [np.sum(df["Cash_Flows"])],
    "PV_Factor": [np.nan],
    "Present_Value": [np.sum(df["Present_Value"])]
})

# Add total row to dataframe
total_df = pd.concat([df, total_row], ignore_index=True)

# Set Decimals to 2 and 4 places as applicable
total_df["Cash_Flows"] = total_df["Cash_Flows"].map("{:,.2f}".format) 
total_df["PV_Factor"] = total_df["PV_Factor"].map("{:,.4f}".format)
total_df["Present_Value"] = total_df["Present_Value"].map("{:,.2f}".format)

# Display the dataframe
display(HTML(total_df.to_html(index=False)))

Years,Cash_Flows,PV_Factor,Present_Value
0,"-50,000.00",1.0000,"-50,000.00"
1,"25,000.00",0.9346,"23,364.49"
2,"20,000.00",0.8734,"17,468.77"
3,"15,000.00",0.8163,"12,244.47"
Totals,"10,000.00",nan,"3,077.73"


In [41]:
# Just to make sure
npv = npf.npv(r, struct['Cash_Flows'])
print(f'Net Present Value = {npv:,.2f}')

Net Present Value = 3,077.73


## Example 4.12 - Effective Annual Rates

What is the end-of-year wealth if Jane Christine receives a stated annual interest rate of 24 percent compounded monthly on a 1 dollar investment?

Using Equation 4.6: 

*C0 * (1 + (r / m))^m*  

In [44]:
r = .24
m = 12
investment = 1

wealth = investment * (1 + (r / m))**m
print(f"her wealth is: ${wealth:.4f}")

her wealth is: $1.2682


The annual rate of return is 26.82 percent.  This annual rate of return is called either the **effective annual rate (EAR)** or the **effective annual yield (EAY).**  Due to compounding, the effective annual interest rate is grater than the stated annual interest rate of 24 percent. Algebraically, we can rewrite the effective annual interest rate as follows:

**Effective Annual Rate**:

*EAR = (1 + r / m)^m - 1*

## Example 4.13 - Compounding Frequencies

If the stated annual rate of interest, 8 percent, is compounded quarterly, what is the effective annual rate?

In [50]:
r = .08
m = 4

EAR = (1 + (r / m))**m - 1
print(f'Thea Effective Annual Rate is: {EAR:.4f} = {EAR*100:.2f}%')

Thea Effective Annual Rate is: 0.0824 = 8.24%


## Example 4.14 - Multiyear Compounding

Harry DeAngelo is investing 5,000 at a stated annual interest rate of 12 percent per year, compounded quarterly, for five years.  What is his wealth at the end of five years?

In [54]:
r = .12
m = 4
n = 5
investment = 5000
wealth = investment * (1 + r / m)**(m*n)
print(f'His wealth will be ${wealth:,.2f} at the end of 5 years.')

His wealth will be $9,030.56 at the end of 5 years.


## Example 4.15 - Continuous Compounding

Linda DeFond invested 1,000 at a continuously compounded rate of 10 percent for one year.  What is the value of her wealth at the end of one year?

**Equation 4.9**

*C0 x e^rT*

In [57]:
n = 1
r = .10
e = 2.718
investment = 1000

wealth = investment * (e**(r*n))
print(f'Her wealth at the end of one year is ${wealth:,.2f}')

Her wealth at the end of one year is $1,105.16


## Example 4.16 - Continuous Compounding, Continued

Linda DeFond's brother, Mark, invested 1,000 at a continuously compoudne rate of 10 percent for two years.  What is the value of his wealth at the end of two years?

In [58]:
n = 2
r = .10
e = 2.718
investment = 1000
wealth = investment * (e**(r*n))
print(f'His wealth at the end of two years is ${wealth:,.2f}')

His wealth at the end of two years is $1,221.38


## Exercise 4.17 - Present Value with Continuous Compounding

The Michigan State Lottery is going to pay you 100,000 at the end of four years.  If the annual continuously compounded rate of interest is 8 percent, what is the present value of this payment?

In [61]:
n = 4
r = .08
e = 2.718
payment = 100000
pv = payment * (1 / (e**(r*n)))
print(f'The present value of the payment is ${pv:,.2f}')

The present value of the payment is $72,617.31


## Example 4.18 - Perpetuities

Consider a perpetuity paying 100 dollars a year.  If the relevant interest rate is 8 percent, what is the value of the consol?

**Equation 4.10**

*PV = C / r*

In [5]:
r = .08
pmt = 100
pv = pmt / r
print(f"The value of the consol is: ${pv:,.2f}")


The value of the consol is: $1,250.00


Now suppose that interest rates fall to 6 percent.  Using equation 4.10 the value of the perpetuity is:

In [6]:
r = .06
pv = pmt / r
print(f"At 6% the value of the consol is: ${pv:,.2f}")


At 6% the value of the consol is: $1,666.67


Note that the value of the perpetuity rises with a drop in the interest rate.  Vonversely, teh value of the perpetuity falls with a rise in the interest rate.

## Example 4.19 - Paying Dividends

Popovich corporation is just about to pay a dividend of 3.00 per share.  Investors anticipate that the annual dividend will rise by 6 percent a year forever.  The applicable discount rate is 11 percent.  What is the price of the stock today?

**Equation 4.12**
*PV = C / ( r - g)*

The numerator in Equation 4.12 is the cash flow to be received net period.  Since the growth rate is 6 percent, the dividend next year is:

In [16]:
r = .11
g = .06
current_dividend = 3
future_dividend = current_dividend * (1 + g)
print(f"${future_dividend:.2f}")

$3.18


The price of the stock today is:

In [17]:
pv_of_divs = future_dividend / (r - g)
stock_price = current_dividend + pv_of_divs
print(f"${stock_price:.2f}")

$66.60


The price of 66.60 includes both the dividend to be received immediately and the present value of all dividends beginning a year from now.  Equation 4.12 makes it possible to calculate only the present value of all dividends beginning a year from now.  Be sure you understand this example; test questions on this subject always seem to trip up a few of our students.

## Example 4.20 - Lottery Valuation

mark Young has just won the state lottery, paying 50,000 dollars a year for 20 years.  He is to receive his first payment a year from now.  The state advertises this as the *Million Dollar Lotery* because 1,000,000 = 50,000 x 20.  If the interest rate is 8 percent, what is the present value of the lottery?

In [ ]:
"""
Equation 4.15

                               [             1          ]
Present Value of Annuity = C * [ 1  -  --------------   ]
                               [        ( 1 + r) ^ T    ]
                               [ --------------------   ]
                               [           r            ]
"""

r = .08
n = 20
fv = 0
pmt = 50000
annuity_pv = -npf.pv(fv=fv, rate=r, nper=n, pmt=pmt, when="end")
print(f"The present value of the lottery is: ${annuity_pv:,.2f}")

Rather than being overjoyed at winning, Mr. Young sues the state for misrepresentation and fraud.  His legal brief states that he was promised 1 million but received only 490,907.

## Example 4.21 - Retirement Investing

Suppose you put 3,000 dollars per year into a Roth IRA.  The account pays 6 percent interest per year.  how much will you have when you retire in 30 years?

This question asks for the future value of an annuity of 3,000 dollars per year for 30 years at 6 percent, which we can calculate as follows:

In [28]:
r = .06
n = 30
pmt = 3000
pv = 0
annuity_fv = -npf.fv(pv=pv, rate=r, nper=n, pmt=pmt)
print(f"${annuity_fv:,.2f}")

$237,174.56


So, you'll have close to a quarter million dollars in the account.